pkg/docker/build.go

```diff
@@ -21,9 +25,37 @@ RUN curl -o docker.tgz https://download.docker.com/linux/static/stable/x86_64/do
 COPY . $BUNDLE_DIR
 `
 
+// BuildInput represents stdin passed to the mixin for the build command.
+type BuildInput struct {
+	Config MixinConfig
+}
+
+// MixinConfig represents configuration that can be set on the docker mixin in porter.yaml
+// mixins:
+// - docker:
+//	  version: 19.03.8
+type MixinConfig struct {
+	ClientVersion string `yaml:"clientVersion,omitempty"`
+}
+
 // Build will generate the necessary Dockerfile lines
 // for an invocation image using this mixin
 func (m *Mixin) Build() error {
-	fmt.Fprintf(m.Out, dockerfileLines)
+	var input BuildInput
+	err := builder.LoadAction(m.Context, "", func(contents []byte) (interface{}, error) {
+		err := yaml.Unmarshal(contents, &input)
+		return &input, err
+	})
+	if err != nil {
+		return err
+	}
+
+	if input.Config.ClientVersion != "" {
+		m.DockerVersion = input.Config.ClientVersion
+	}
+
+	log.Println(m.Out, dockerfileLines, m.DockerVersion, m.WorkingDir)
```

Comment:
I believe we want this to print to `m.Out` and it should be updated to use `printf` in tandem with the preceding note (adding a `%s` formatter).  So, something like:

`fmt.Fprintf(m.Out, dockerfileLines, m.DockerVersion)`

(Note, I also dropped `m.WorkingDir` as it doesn't currently apply for this mixin.)

Reply:
got it!